### ISSUE TITLE
building v0.13.0 or v0.14.0 is missing mixin-utils/utils.libsonnet

### ISSUE THREAD

**Issue by desertmoose706** [prometheus-operator/kube-prometheus#2516]
Recently having an issue with building and deploying kube-prometheus 0.13 or 0.14, stating that it is not able to import mixin-utils/utils.libsonnet. I see a bug that was reported and closed (https://github.com/prometheus-operator/kube-prometheus/issues/2463).

The suggestion in that issue as well as the merged fix doesn't work in my case, so I'm curious if I'm doing something wrong with my jsonnet config that would still cause an issue with the missing mixin-utils?

**ERROR:**
```
RUNTIME ERROR: couldn't open import "mixin-utils/utils.libsonnet": no match locally or in the Jsonnet library paths
	vendor/github.com/grafana/jsonnet-libs/grafana-builder/grafana.libsonnet:1:15-51	thunk <utils> from <$>
	vendor/github.com/grafana/jsonnet-libs/grafana-builder/grafana.libsonnet:335:29-34	
	vendor/github.com/grafana/jsonnet-libs/grafana-builder/grafana.libsonnet:346:12-27	object <anonymous>
	vendor/github.com/brancz/kubernetes-grafana/grafana/grafana.libsonnet:[100]:25-79	object <anonymous>
	Field "k8s-resources-cluster.json"	
	Field "data"	
	Array element 9	
	Field "items"	
	Field "grafana-dashboardDefinitions"	
	During manifestation
```

**jsonnet file:**
```
local grafana_fqdn = "k8s-grafana.examplehost.net";
local prometheus_fqdn = "k8s-grafana.examplehost.net";
local alertmanager_fqdn = "k8s-grafana.examplehost.net";

local kp =
  (import 'kube-prometheus/main.libsonnet') +
  (import 'kube-prometheus/addons/all-namespaces.libsonnet') + 
  (import 'kube-prometheus/addons/node-ports.libsonnet') +
  (import 'kube-prometheus/addons/networkpolicies-disabled.libsonnet') +
  {
    values+:: {
      common+: {
        namespace: 'kubeprom',
        platform: 'kubeadm',
      },

      // kubePrometheus+: {
      //   platform: 'eks',
      // },

      alertmanager+:: {
        config: importstr 'alertmanager-config.yaml',
      },

      prometheus+:: {
        namespaces+: ['dev03','dev02','dev01'],
      },

      kubeStateMetrics+:: {
        baseCPU: '200m',
        cpuPerNode: '4m',
        baseMemory: '300Mi',
        memoryPerNode: '60Mi',
      },

      grafana+:: {
        resources: {
          requests: { cpu: '100m', memory: '200Mi' },
          limits: { cpu: '500m', memory: '1500Mi' },
        },
        config: {  
          // http://docs.grafana.org/installation/configuration/
          sections: {
            // Do not require grafana users to login/authenticate
            'auth.anonymous': {
              enabled: true
            },
            server+: {
              root_url: "http://" + grafana_fqdn + "/",
            },
          },
        },
        dashboards+:: {
          'hystrix-dashboard.json': (import '../dashboards/hystrix-dashboard.json'),
          'hystrixv2-dashboard.json': (import '../dashboards/hystrixv2-dashboard.json'),
          'hystrixv2-sb2-dashboard.json': (import '../dashboards/hystrixv2-sb2-dashboard.json'),
          'apache-dashboard.json': (import '../dashboards/apache-dashboard.json'),
        },
      },
    },

    prometheus+:: {
      prometheus+: {
        // Reference info: https://coreos.com/operators/prometheus/docs/latest/api.html#prometheusspec
        spec+: {
          // An e.g. of the purpose of this is so the "Source" links on http://<alert-manager>/#/alerts are valid.
          externalUrl: "http://" + prometheus_fqdn + "/prometheus",
          retention: '30d',
          storage: {
            volumeClaimTemplate: {
              apiVersion: 'v1',
              kind: 'PersistentVolumeClaim',
              spec: {
                accessModes: ['ReadWriteOnce'],
                // https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.11/#resourcerequirements-v1-core (defines 'requests'),
                // and https://kubernetes.io/docs/concepts/policy/resource-quotas/#storage-resource-quota (defines 'requests.storage')
                resources: { requests: { storage: '100Gi' } },
                // A StorageClass of the following name (which can be seen via `kubectl get storageclass` from a node in the given K8s cluster) must exist prior to kube-prometheus being deployed.
                storageClassName: 'prometheus',
                // The following 'selector' is only needed if you're using manual storage provisioning (https://github.com/coreos/prometheus-operator/blob/master/Documentation/user-guides/storage.md#manual-storage-provisioning).
                // And note that this is not supported/allowed by AWS - uncommenting the following 'selector' line (when deploying kube-prometheus to a K8s cluster in AWS) will cause the pvc to be stuck in the Pending status and have the following error:
                //  * 'Failed to provision volume with StorageClass "ssd": claim.Spec.Selector is not supported for dynamic provisioning on AWS'
                #selector: { matchLabels: { env: 'monitoring', app: 'prometheus' } },
              },
            },
          },
        },
      },
    },

    alertmanager+:: {
      alertmanager+: {
        // Reference info: https://github.com/coreos/prometheus-operator/blob/master/Documentation/api.md#alertmanagerspec
        spec+: {
          externalUrl: "http://" + alertmanager_fqdn + "/alertmanager",
        },
      },
    },
    
  };

//{ '0namespace-namespace': kp.kubePrometheus.namespace } +
{
  ['setup/prometheus-operator-' + name]: kp.prometheusOperator[name]
  for name in std.filter((function(name) name != 'serviceMonitor' && name != 'prometheusRule'), std.objectFields(kp.prometheusOperator))
} +
// { 'setup/pyrra-slo-CustomResourceDefinition': kp.pyrra.crd } +
// serviceMonitor and prometheusRule are separated so that they can be created after the CRDs are ready
{ 'prometheus-operator-serviceMonitor': kp.prometheusOperator.serviceMonitor } +
{ 'prometheus-operator-prometheusRule': kp.prometheusOperator.prometheusRule } +
{ 'kube-prometheus-prometheusRule': kp.kubePrometheus.prometheusRule } +
{ ['alertmanager-' + name]: kp.alertmanager[name] for name in std.objectFields(kp.alertmanager) } +
{ ['blackbox-exporter-' + name]: kp.blackboxExporter[name] for name in std.objectFields(kp.blackboxExporter) } +
{ ['grafana-' + name]: kp.grafana[name] for name in std.objectFields(kp.grafana) } +
// { ['pyrra-' + name]: kp.pyrra[name] for name in std.objectFields(kp.pyrra) if name != 'crd' } +
{ ['kube-state-metrics-' + name]: kp.kubeStateMetrics[name] for name in std.objectFields(kp.kubeStateMetrics) } +
{ ['kubernetes-' + name]: kp.kubernetesControlPlane[name] for name in std.objectFields(kp.kubernetesControlPlane) }
{ ['node-exporter-' + name]: kp.nodeExporter[name] for name in std.objectFields(kp.nodeExporter) } +
{ ['prometheus-common' + name]: kp.prometheus[name] for name in std.objectFields(kp.prometheus) } +
{ ['prometheus-adapter-' + name]: kp.prometheusAdapter[name] for name in std.objectFields(kp.prometheusAdapter) }
//{ [name + '-ingress']: kp.ingress[name] for name in std.objectFields(kp.ingress) }

```

build script:
```
#!/bin/bash
# This script uses arg $1 (name of *.jsonnet file to use) to generate the manifests/*.yaml files.

set -e
set -x
# only exit with zero if all commands of the pipeline exit successfully
set -o pipefail

# Make sure to start with a clean 'manifests' dir
rm -rf ${2}/manifests
mkdir -p ${2}/manifests
mkdir -p ${2}/manifests/setup

# optional, but we would like to generate yaml, not json
jsonnet -J vendor -m ${2}/manifests "${1-example.jsonnet}" | xargs -I{} sh -c 'cat {} | gojsontoyaml > {}.yaml; rm -f {}' -- {}
#jsonnet -J vendor -m manifests "${1-example.jsonnet}"
```

build environment setup in pipeline:
```
  # install GO
  - apk add go
  - apk add gettext
  - apk add git
  - git config --global user.name "Name Name"
  - git config --global user.email "name.name@examplehost.com"
  # install jsonnet stuff
  - export PATH=${PATH}:`go env GOPATH`/bin
  - go install -a github.com/jsonnet-bundler/jsonnet-bundler/cmd/jb@latest
  - go install github.com/brancz/gojsontoyaml@latest
  - go install github.com/google/go-jsonnet/cmd/jsonnet@latest
  # install jb
  - jb init
  - jb install github.com/prometheus-operator/kube-prometheus/jsonnet/kube-prometheus@${KUBE_RELEASE}
  # build manifest files
  - ./build.sh ${CLUSTER}/${CLUSTER}.jsonnet ${CLUSTER}
```

**Comment by desertmoose706** [comment]
Not sure if this is just my environment, but the fix for me was to download the repo that contains mixin-utils via:
`jb install github.com/grafana/jsonnet-libs`

and then update the build.sh command to include the path to where the mixin-utils is to the jsonnet command:
`jsonnet -J vendor -J ./vendor/github.com/grafana/jsonnet-libs -m ${2}/manifests "${1-example.jsonnet}" | xargs -I{} sh -c 'cat {} | gojsontoyaml > {}.yaml; rm -f {}' -- {}`

So, this must have been a jsonnet pathing issue for me, but honestly not sure why it started because I've used this same pipelnie to deploy kube-prom in the past, maybe a few months ago.

**Comment by desertmoose706** [comment]
Final comments. I updated the container image to Ubuntu from Alpine. Also here are the updates as a whole:

**build.sh**
```
#!/bin/bash
# This script uses arg $1 (name of *.jsonnet file to use) to generate the manifests/*.yaml files.

set -e
set -x
# only exit with zero if all commands of the pipeline exit successfully
set -o pipefail

# Make sure to start with a clean 'manifests' dir
rm -rf ${2}/manifests
mkdir -p ${2}/manifests
mkdir -p ${2}/manifests/setup

# optional, but we would like to generate yaml, not json
# jsonnet -J vendor -m ${2}/manifests "${1-example.jsonnet}" | xargs -I{} sh -c 'cat {} | gojsontoyaml > {}.yaml; rm -f {}' -- {}
jsonnet -J vendor -J ./vendor/github.com/grafana/jsonnet-libs -m ${2}/manifests "${1-example.jsonnet}" | xargs -I{} sh -c 'cat {} | gojsontoyaml > {}.yaml; rm -f {}' -- {}
#jsonnet -J vendor -m manifests "${1-example.jsonnet}"
```

**pipeline**
```
  # install GO
  - apt-get -y install golang-go
  - git config --global user.name "Name Name"
  - git config --global user.email "name.name@example.com"
  # install jsonnet stuff
  - export PATH=${PATH}:`go env GOPATH`/bin
  - go install -a github.com/jsonnet-bundler/jsonnet-bundler/cmd/jb@latest
  - go install github.com/brancz/gojsontoyaml@latest
  - go install github.com/google/go-jsonnet/cmd/jsonnet@latest
  # install jb
  - jb init
  - jb install github.com/prometheus-operator/kube-prometheus/jsonnet/kube-prometheus@${KUBE_RELEASE}
  - jb install github.com/grafana/jsonnet-libs
  # build manifest files
  - ./build.sh ${CLUSTER}/${CLUSTER}.jsonnet ${CLUSTER}
```

**Comment by briend** [comment]
This happens to me as well on `0.12` (in the past this release was fine) as well as `0.14`.  I have no problems at all when using the `main` branch, so I think there is some problem that will be resolved in `0.15` when that is released
